lib/blemulator/internal/simulation_manager_mixins/characteristics_mixin.dart

```diff
@@ -59,7 +59,7 @@ mixin CharacteristicsMixin on SimulationManagerBase {
       if (characteristic != null) break peripheralsLoop;
     }
 
-    _throwErrorIfNull(targetCharacteristic);
+    _throwErrorIfNull(targetCharacteristic, characteristicUUID);
```

Comment:
This even works? targetCharacteristic is never set.

Reply:
It was a bug which was fixed in monitoring characteristics branch. I fixed it here too.